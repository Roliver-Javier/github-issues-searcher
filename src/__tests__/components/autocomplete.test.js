import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';
import Autocomplete from '../../components/autocomplete/autocomplete';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';

configure({ adapter: new Adapter() });
const mockStore = configureStore([]);
const renderComponent = (store, props = {}) =>
  TestRenderer.create(
    <Provider store={store}>
      <Autocomplete {...props} />
    </Provider>
  ).toJSON();
const initialState = {
  issues: {
    data: [],
    dataFiltered: [],
    issueTyped: '',
  },
};
describe('autocomplete Component', () => {
  let store;
  let wrapper;
  let dispatchMock;
  beforeEach(() => {
    store = mockStore({
      ...initialState,
    });
    dispatchMock = jest.fn();
  });
  it('renders the autocomplete', () => {
    wrapper = renderComponent(store, {
      dispatch: dispatchMock,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
