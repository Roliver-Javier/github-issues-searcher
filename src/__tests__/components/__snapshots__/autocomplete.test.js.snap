// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`autocomplete Component renders the autocomplete 1`] = `
<div
  className="container"
  data-test="container"
>
  <section
    className="autocompleteFilter"
  >
    <input
      className="primaryTxt"
      data-test="input-autocomplete"
      onChange={[Function]}
      placeholder="Type Here"
      type="text"
      value=""
    />
  </section>
  <section
    className="panel container"
  />
</div>
`;
